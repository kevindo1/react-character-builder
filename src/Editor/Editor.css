img {
  display: flex;
  flex-direction: column;
  margin: auto;
  height: 200px;
  width: 200px;
}

body {
  color: white;
}
