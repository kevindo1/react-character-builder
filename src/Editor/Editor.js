import React from 'react';
import './Editor.css';

export default function Editor({
  head,
  setHead,
  middle,
  setMiddle,
  bottom,
  setBottom,
  phrase,
  setHeadCount,
  setMiddleCount,
  setBottomCount,
  setPhraseList,
  setPhrase,
}) {
  const handleHead = (e) => {
    setHead(e.target.value);
    setHeadCount((prevState) => prevState + 1);
  };
  const handleMiddle = (e) => {
    setMiddle(e.target.value);
    setMiddleCount((prevState) => prevState + 1);
  };
  const handleBottom = (e) => {
    setBottom(e.target.value);
    setBottomCount((prevState) => prevState + 1);
  };
  const handleClick = () => {
    setPhraseList((prevState) => [...prevState, phrase]);
    setPhrase('');
  };

  return (
    <div className="editor">
      <select value={head} onChange={handleHead}>
        <option value="bird-head">{'Bird'}</option>
        <option value="duck-head">Duck</option>
        <option value="dog-head">Dog</option>
        <option value="horse-head">Horse</option>
      </select>
      <select value={middle} onChange={handleMiddle}>
        <option value="blue-middle">{'Blue'}</option>
        <option value="dress-middle">Fancy</option>
        <option value="pink-middle">Pink</option>
        <option value="red-middle">Red</option>
      </select>
      <select value={bottom} onChange={handleBottom}>
        <option value="leg-pants">{'Single Leg'}</option>
        <option value="white-pants">White Pants</option>
        <option value="blue-pants">Blue Jeans</option>
      </select>
      <input type="text" value={phrase} onChange={(e) => setPhrase(e.target.value)} />
      <button className="btn" onClick={handleClick}>
        Add Catchphrase
      </button>
    </div>
  );
}
