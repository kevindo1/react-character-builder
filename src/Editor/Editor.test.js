import { render } from '@testing-library/react';
import Editor from './Editor';

test('renders the Editor panel', () => {
  const container = render(<Editor head="bird-head" />);
  expect(container).toMatchSnapshot();
});
