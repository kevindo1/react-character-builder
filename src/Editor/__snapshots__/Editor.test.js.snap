// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Editor panel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="editor"
      >
        <select>
          <option
            value="bird-head"
          >
            Bird
          </option>
          <option
            value="duck-head"
          >
            Duck
          </option>
          <option
            value="dog-head"
          >
            Dog
          </option>
          <option
            value="horse-head"
          >
            Horse
          </option>
        </select>
        <select>
          <option
            value="blue-middle"
          >
            Blue
          </option>
          <option
            value="dress-middle"
          >
            Fancy
          </option>
          <option
            value="pink-middle"
          >
            Pink
          </option>
          <option
            value="red-middle"
          >
            Red
          </option>
        </select>
        <select>
          <option
            value="leg-pants"
          >
            Single Leg
          </option>
          <option
            value="white-pants"
          >
            White Pants
          </option>
          <option
            value="blue-pants"
          >
            Blue Jeans
          </option>
        </select>
        <input
          type="text"
          value=""
        />
        <button
          class="btn"
        >
          Add Catchphrase
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="editor"
    >
      <select>
        <option
          value="bird-head"
        >
          Bird
        </option>
        <option
          value="duck-head"
        >
          Duck
        </option>
        <option
          value="dog-head"
        >
          Dog
        </option>
        <option
          value="horse-head"
        >
          Horse
        </option>
      </select>
      <select>
        <option
          value="blue-middle"
        >
          Blue
        </option>
        <option
          value="dress-middle"
        >
          Fancy
        </option>
        <option
          value="pink-middle"
        >
          Pink
        </option>
        <option
          value="red-middle"
        >
          Red
        </option>
      </select>
      <select>
        <option
          value="leg-pants"
        >
          Single Leg
        </option>
        <option
          value="white-pants"
        >
          White Pants
        </option>
        <option
          value="blue-pants"
        >
          Blue Jeans
        </option>
      </select>
      <input
        type="text"
        value=""
      />
      <button
        class="btn"
      >
        Add Catchphrase
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
